File: MasterDirectory/MasterDirectory.Web/Modules/ServerTypes/Digital/CategoriaDigitalForm.ts
import { StringEditor, PrefixedContext } from "@serenity-is/corelib";
import { initFormType } from "@serenity-is/corelib/q";

export interface CategoriaDigitalForm {
    LocalSap: StringEditor;
    DirGoogle: StringEditor;
    Latitud: StringEditor;
    Longitud: StringEditor;
}

export class CategoriaDigitalForm extends PrefixedContext {
    static formKey = 'Digital.CategoriaDigital';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!CategoriaDigitalForm.init)  {
            CategoriaDigitalForm.init = true;

            var w0 = StringEditor;

            initFormType(CategoriaDigitalForm, [
                'LocalSap', w0,
                'DirGoogle', w0,
                'Latitud', w0,
                'Longitud', w0
            ]);
        }
    }
}

File: MasterDirectory/MasterDirectory.Web/Modules/Digital/CategoriaDigital/CategoriaDigitalForm.cs
using Serenity.ComponentModel;
using System;

namespace MasterDirectory.Digital.Forms;

[FormScript("Digital.CategoriaDigital")]
[BasedOnRow(typeof(CategoriaDigitalRow), CheckNames = true)]
public class CategoriaDigitalForm
{
    [Tab("General")]
    public string LocalSap { get; set; }
    public string DirGoogle { get; set; }
    public string Latitud { get; set; }
    public string Longitud { get; set; }
    public string LinkGoogleMaps { get; set; }

    //public DateTime DtRegistro { get; set; }
}